spec.md

```diff
@@ -0,0 +1,81 @@
+---
+layout: page
+title: Semantic Specification
+---
+
+## Concepts and Terminology
+
+A **Trace** represents the potentially distributed, potentially concurrent data/execution path in a (potentially distributed, potentially concurrent) system.
+
+A **Span** represents a logical unit of work in the system that has a start time and an end time. Spans may be nested and ordered to model parent-child and casual relationships. A Trace can be thought of as a tree of Spans.
+
+Every Span has zero or more **Logs**, each of which being a timestamped message with an optional structured data payload of arbitrary size.
+
+Every Span may also have zero or more key/value **Tags**, which do not have timestamps and simply annotate the spans.
+
+A **Trace Context** encapsulates the smallest amount of state needed to describe a Span's identity within a larger, potentially distributed trace, sufficient to propagate the context of a particular trace between processes.
+
+**Trace Attributes** are key/value pairs stored in a Trace Context and propagated _in-band_ to all future child Spans. Given a full-stack OpenTracing integration, Trace Attributes enable powerful functionality by transparently propagating arbitrary application data: e.g., from a mobile app all the way into the depths of a storage system. Trace attributes come with powerful _costs_ as well; since the attributes are propagated in-band, if they are too large they can reduce system throughput or contribute to RPC latency. Use Trace Attributes with care.
+
+**Trace Attributes** vs. **Span Tags**
+
+* Trace Attributes are propagated in-band across process boundaries. Span Tags are not propagated.
+* Span Tags are recorded off-band in the tracing system's storage. Trace Attributes are not recorded.
+
+## Platform-Independent API Semantics
+
+OpenTracing supports a number of different platforms, and of course the per-platform APIs try to blend in to their surroundings and do as the Romans do. That said, each platform API must model a common set of semantics for the core tracing concepts described above. In this document we attempt to describe those concepts and semantics in a language- and platform-agnostic fashion.
+
+#### Tracer
+
+The `Tracer` interface must have the following capabilities:
+
+- Start a `Span` that has no parent **(py: `start_trace`, go: `StartTrace`)**
+- Start a `Span` given a parent `TraceContext` **(py: `join_trace`, go: `JoinTrace`)**
+- Start a `Span` associated with a specific `TraceContext` **(py: `start_span_with_context`, go: `StartSpanWithContext`)**
+
+
+#### Span
+
+The `Span` interface must have the following capabilities:
+
+- Create and start a new child `Span` with a given operation name **(py: `start_child`, go: `StartChild`)**
```

Comment:
Two cents: it might be nice to explicitly spell out here what *must* happen and what is implementation dependent with regards to the inheritance from parent to child of both trace attributes and tags.